### PR TITLE
feat(STONEINTG-967): consider 0 ReleasePlans in 'AutoReleased' field

### PR DESCRIPTION
**Pull request by dheerajodha**
Before this commit:
* If there were 0 ReleasePlans available when a a Snapshot passed all of its required tests or there were 0 tests, the Snapshot's "AutoRelease" condition type's message field would contain the text "The Snapshot was auto-released".
* Which is misleading as the Snapshot couldn't be released due to 0 ReleasePlans available.

After this commit:
* We will check the count of ReleasePlans and if its 0 then we would still mark snapshot as AutoRelease but make sure to mention this in its message field.
* This would help users who are wondering why there are no release(s) even though the snapshot is marked as AutoReleased.

## Maintainers will complete the following section

- [ ] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [ ] Code coverage from testing does not decrease and new code is covered ([check the PR coverage on codecov](https://app.codecov.io/gh/redhat-appstudio/integration-service/pulls))
- [ ] [Controllers diagrams](https://github.com/konflux-ci/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

### CODE REVIEW

**Review by chipspeak — APPROVED** [review]
LGTM

**Review by Josh-Everett — APPROVED** [review]

**Comment by dheerajodha** [comment]
FYI: The latest commit adds unit test coverage for this change.

**Review by 14rcole — APPROVED** [review]

File: internal/controller/snapshot/snapshot_adapter.go [hunk 1]
```diff
@@ -406,25 +408,29 @@ func (a *Adapter) EnsureAllReleasesExist() (controller.OperationResult, error) {
...
-			return a.RequeueIfYoungerThanThreshold(errors.Join(err, er))
+	if len(*releasePlans) > 0 {
+		err = a.createMissingReleasesForReleasePlans(a.application, releasePlans, a.snapshot)
+		if err != nil {
+			a.logger.Error(err, "Failed to create new Releases")
+			patch := client.MergeFrom(a.snapshot.DeepCopy())
+			gitops.SetSnapshotIntegrationStatusAsError(a.snapshot, "Failed to create new Releases: "+err.Error())
+			a.logger.LogAuditEvent("Snapshot integration status marked as Invalid. Failed to create new Releases",
+				a.snapshot, h.LogActionUpdate)
+			er := a.client.Status().Patch(a.context, a.snapshot, patch)
```
Comment (kasemAlem) [inline internal/controller/snapshot/snapshot_adapter.go:419]:
```suggestion
			patchErr := a.client.Status().Patch(a.context, a.snapshot, patch)
```

Comment (dheerajodha) [reply internal/controller/snapshot/snapshot_adapter.go]: thanks! updated

**Review by kasemAlem — APPROVED** [review]

**Comment by kasemAlem** [comment]
LGTM

**Pull request merged**